openedx_webhooks/tasks/utils.py

```diff
@@ -0,0 +1,118 @@
+"""
+from openedx_webhooks.tasks import logger
+
+
+def _log(log_fn, task_request, message):
+    """
+    Logs a message using the supplied logging function.
+
+    Args:
+        log_fn: logging.Logger instance method ready to accept log message.
+        task_request (Task.request): Celery task instance `request`_ object.
+        message (str): Log message.
+
+    Returns:
+        None
+
+    .. _request:
+        http://docs.celeryproject.org/en/latest/userguide/tasks.html#context
+    """
+    log_fn("{}: {}".format(task_request.id, message))
+
+
+def log_info(task_request, message):
+    """
+    Logs a message using the Celery's app logger at the INFO level.
+
+    Args:
+        task_request (Task.request): Celery task instance `request`_ object.
+        message (str): Log message.
+
+    Returns:
+        None
+
+    .. _request:
+        http://docs.celeryproject.org/en/latest/userguide/tasks.html#context
+    """
+    _log(logger.info, task_request, message)
+
+
+def log_error(task_request, message):
+    """
+    Logs a message using the Celery's app logger at the ERROR level.
+
+    Args:
+        task_request (Task.request): Celery task instance `request`_ object.
+        message (str): Log message.
+
+    Returns:
+        None
+
+    .. _request:
+        http://docs.celeryproject.org/en/latest/userguide/tasks.html#context
+    """
+    _log(logger.error, task_request, message)
+
+
+def log_request(task_request, request):
+    """
+    Logs a message using the Celery's app logger at the INFO level for POST
+    requests.
+
+    Args:
+        task_request (Task.request): Celery task instance `request`_ object.
+        request (requests.PreparedRequest): `requests.PreparedRequest`_ object.
+
+    Returns:
+        None
+
+    .. _request:
+        http://docs.celeryproject.org/en/latest/userguide/tasks.html#context
+    .. _requests.PreparedRequest:
+        http://docs.python-requests.org/en/master/api/#requests.PreparedRequest
+    """
+    msg = "{0.method} {0.url}: {0.body}".format(request)
+    log_info(task_request, msg)
+
+
+def log_response(task_request, response):
+    """
+    Logs a message using the Celery's app logger at the INFO level for response
+
+    Args:
+        task_request (Task.request): Celery task instance `request`_ object.
+        response (requests.Response): A `requests.Response`_ object.
+
+    Returns:
+        None
+
+    .. _request:
+        http://docs.celeryproject.org/en/latest/userguide/tasks.html#context
+    .. _requests.Response:
+        http://docs.python-requests.org/en/master/api/#requests.Response
+    """
+    msg = "{status_code} {reason} for {url}: {content}".format(**response)
+    log_info(task_request, msg)
+
+
+def log_request_response(task_request, response):
+    """
+    Logs two messages using the Celery's app logger at the INFO level for
+    request and response.
+
+    Args:
+        task_request (Task.request): Celery task instance `request`_ object.
```

Comment:
My instinct was the same, but again, it wasn't clear to me what the edX standards are. I thought I would err on the side of more formal documentation.